fig/backends/aws/__init__.py

```diff
@@ -1,50 +1,153 @@
+                log.info("Detection already submitted to Security Hub. Alert not processed.")
+            else:
+                if response["SuccessCount"] > 0:
+                    submit_msg = f"Detection submitted to Security Hub. (Request ID: {response['ResponseMetadata']['RequestId']})"
+                    log.info(submit_msg)
 
     def create_payload(self):
+        region = config.get('aws', 'region')
+        try:
+            account_id = boto3.client("sts").get_caller_identity().get('Account')
+        except KeyError:
+            # Failed to get endpoint_resolver the first time, try it again
+            account_id = boto3.client("sts").get_caller_identity().get('Account')
+        severity_product = self.event.severity_value
+        severity_normalized = severity_product * 20
         payload = {
-            "hostname": self.event.device_details["hostname"],
-            "instance_id": self.event.instance_id,
-            "service_provider_account_id": self.event.cloud_provider_account_id,
-            "local_ip": self.event.device_details["local_ip"],
-            "mac_address": self.event.device_details["mac_address"],
-            "detected_mac_address": self.event.original_event["event"]["MACAddress"],
-            "detected_local_ip": self.event.original_event["event"]["LocalIP"],
-            "detection_id": self.event.event_id,
-            "tactic": self.event.original_event["event"]["Tactic"],
-            "technique": self.event.original_event["event"]["Technique"],
-            "description": self.event.detect_description,
-            "source_url": self.event.falcon_link,
-            "generator_id": "Falcon Host",
-            "types": ["Namespace: Threat Detections"],
-            "created_at": self.event.time,
-            "updated_at": ((datetime.utcfromtimestamp(datetime.timestamp(datetime.now()))).isoformat() + 'Z'),
-            "record_state": "ACTIVE",
-            "severity": self.event.severity}
+            "SchemaVersion": "2018-10-08",
+            "ProductArn": "arn:aws:securityhub:{}:517716713836:product/crowdstrike/crowdstrike-falcon".format(region),
+            "AwsAccountId": account_id,
+            "SourceUrl": self.event.falcon_link,
+            "GeneratorId": "Falcon Host",
+            "CreatedAt": datetime.utcfromtimestamp(float(self.event.event_create_time) / 1000.).isoformat() + 'Z',
+            "UpdatedAt": ((datetime.utcfromtimestamp(datetime.timestamp(datetime.now()))).isoformat() + 'Z'),
+            "RecordState": "ACTIVE",
+            "Severity": {"Product": severity_product, "Normalized": severity_normalized}
+        }
+
+        # Instance ID based detail
+        if self.event.instance_id:
+            payload["Id"] = f"{self.event.instance_id}{self.event.event_id}"
+            payload["Title"] = "Falcon Alert. Instance: %s" % self.event.instance_id
+            payload["Resources"] = [{"Type": "AwsEc2Instnace", "Id": self.event.instance_id, "Region": region}]
```

Comment:
I don't understand this part.

The `region` variable here is taken from the FIG config. It may or may not coincide with the region of the instance?

Not boto3 expert, but in the `find_instance` method it seems we are iterating through the regions to find the instance in its respective region. While here we hard-code the fig region to the payload.

Reply:
The hard-coded region is where the Security Hub finding is reported.